### ISSUE TITLE
Feature request: gradient and drop shadow support in FreeTypeFontGenerator

### ISSUE THREAD

**Issue by notsonotso** [libgdx/libgdx#4900]
There's a 3'rd party version here, but it doesn't compile anymore:

http://www.badlogicgames.com/forum/viewtopic.php?f=17&t=18157

Would be great if this was officially maintained, as doing this in Hiero is a major inflexible hassle compared to generating at runtime.

Thanks for considering.

**Comment by xGnoSiSx** [comment]
This looks cool. But,

Do we have a way of creating Gradients in general? They can have so many uses in texture generation and drawing in general. If you ever used photoshop for gradiends you would understand what I mean. The options and key points for colors and opacity make the potential code too big - e.g. repeating rainbow gradients.

**Comment by Velocity-** [comment]
@xGnoSiSx gradients in OpenGL can be achieved by specifying two colors in a quad.

I also don't see how it makes the code big? No matter how many key points you want to add, it's no more than an array of points, no? I'm not sure I'm following what you're saying here.

Wonder if this could be achieved - maybe somewhat inefficient - by creating a texture, drawing black letters, applying gaussian blur to the whole texture with a given strength and then splatting that behind the main atlas.

**Comment by xGnoSiSx** [comment]
There are radial gradients, gradient rotations and other techniques, like noise types, alpha gradient, textured gradients and on and on.  I wouldn't like this introduced because then there would be requests for more functionality and options - which would be out of scope.

I would welcome the introduction of a a glyph texture backdrop in Hiero. There are myriad choices on letter texturing, just allow the user to supply a texture (image) sized in any diimensions, that will be used for filling the glyphs, tiled or untiled.

In this way more versatility is added with minimal complexity - each user can select their fill texture in the tool.

But i agree on shadows. Last time, I baked them in the font texture myself and it was kind of a waste of time.

**Comment by notsonotso** [comment]
@xGnoSiSx Right. One thing is the waste of time. The other is the lack of flexibility. Would be great to be able to do this at runtime.

**Comment by xGnoSiSx** [comment]
I agree.
